torch/export/_safeguard.py

```diff
@@ -0,0 +1,15 @@
+import torch
+from torch.overrides import TorchFunctionMode
+
+
+class GradModeUnsupportedSafeguard(TorchFunctionMode):
+    def __torch_function__(self, func, types, args=(), kwargs=None):
+        kwargs = kwargs or {}
+        unsupported_grad_mode_ops = [
+            torch._C._set_grad_enabled,
+        ]
+        if func in unsupported_grad_mode_ops and torch._C._get_dispatch_mode(
```

Comment:
Leave comment on how this approach works and why we are checking the mode here? Also cc @bdhirsh

Reply:
Can you rephrase it to just say: "We only enable this mode while tracing by checking the torch dispatch mode to see if there is any active ProxyMode"

Reply:
@tugsbayasgalan just checking - is the idea that this mode will only be turned on when using aot-dispatch export, and it will be off when using pre-dispatch export?

Reply:
Yep!

Reply:
yeah actually good catch! We will need to check if the mode is pre_dispatch or not first.

Reply:
@andrewlee302 I think you should also check if the ProxyTorchDispatchMode is not pre-dispatch before erroring. I think the object has the field already.